src/lib/peer_trust/peer_trust.ml

```diff
@@ -46,15 +47,216 @@ end) (Now : sig
   val now : unit -> Time.t
 end)
 (Action : Action_intf)
-(Db : Key_value_database.S with type key := Peer.t and type value := Record.t) :
-  S with type peer := Peer.t and type action := Action.t = struct
-  type t = unit
+(Db : Key_value_database.S with type key := Peer.t and type value := Record.t) =
+struct
+  type t =
+    { db: Db.t
+    ; bans_reader: Peer.t Strict_pipe.Reader.t
+    ; bans_writer:
+        (Peer.t, Strict_pipe.synchronous, unit Deferred.t) Strict_pipe.Writer.t
+    }
+
+  module Record_inst = Record.Make (Now)
 
-  let create = stub ()
+  let create ~db_dir =
+    let reader, writer = Strict_pipe.create Strict_pipe.Synchronous in
+    {db= Db.create ~directory:db_dir; bans_reader= reader; bans_writer= writer}
 
-  let record = stub ()
+  let ban_pipe {bans_reader} = bans_reader
```

Comment:
comment should indicate purpose of the ban_pipe

Reply:
added a doc comment in the mli